autocorns/biologist.py

```diff
@@ -1,86 +1,115 @@
 import argparse
-from concurrent.futures import as_completed, ThreadPoolExecutor, Future
-import csv
 import datetime
 import json
 import os
 import sys
 import time
-from typing import Any, Dict, List, Optional, Set, Tuple
+from typing import Any, Dict, List, Optional, Tuple
 
-from brownie import network
+
+from brownie import network, web3
 from brownie.network import chain
 import requests
 from tqdm import tqdm
 
 from . import ERC721WithDiamondStorage
 from . import MetadataFacet
+from . import Multicall2
 from . import StatsFacet
 from eth_typing.evm import ChecksumAddress
 
 
+CALL_CHUNK_SIZE = 500
+
+
+Multicall2_address = "0xc8E51042792d7405184DfCa245F2d27B94D013b6"
+
+
+def make_multicall(
+    multicall_method: Any,
+    brownie_contract_method: Any,
+    address: web3.toChecksumAddress,
+    inputs: List[Any],
+    block_number: str = "latest",
+) -> Any:
+    multicall_result = multicall_method.call(
+        False,  # success not required
+        [
+            (
+                address,
+                brownie_contract_method.encode_input(input),
+            )
+            for input in inputs
+        ],
+        block_identifier=block_number,
+    )
+
+    results = []
+
+    # Handle the case with not successful calls
+    for encoded_data in multicall_result:
+        if encoded_data[0]:
+            results.append(brownie_contract_method.decode_output(encoded_data[1]))
+        else:
+            print(encoded_data, file=sys.stderr)
+            results.append(None)
```

Comment:
Why append `None`? Why not just skip this line?

Reply:
None require because we do zip with tokens_ids wich filtered by None.